src/sqlfluff/core/rules/config_info.py

```diff
@@ -46,6 +46,10 @@
             "Run this rule even for dialects where this rule is disabled by default."
         ),
     },
+    "case_sensitive": {
```

Comment:
are any dialects case insensitive? If not, i'm not sure we should offer this config. We don't consider case when we parse

Reply:
To my knowledge, most dialect aren't case sensitive _unless quoted_ and then they are. It's a good point though, that maybe I should just handle quoted and unquoted differently.

Quoted is case sensitive. Unquoted is not. MySQL and postgres fold to lowercase, snowflake folds to uppercase:
- https://dev.mysql.com/doc/refman/8.0/en/identifier-case-sensitivity.html
- https://www.postgresql.org/docs/current/sql-syntax-lexical.html
- https://docs.snowflake.com/en/sql-reference/identifiers-syntax

There's a good argument that while two quoted columns `"foo"` and `"FOO"` might be _valid_ and actually refer to different columns - it's probably still not _recommended_. I'll update the docstring and take away the config.

Reply:
docstring now says:

```python
class Rule_AL08(BaseRule):
    """Column aliases should be unique within each clause.

    Reusing column aliases is very likely a coding error. Note that while
    in many dialects, quoting an identifier makes it case-sensitive
    this rule always compares in a case-insensitive way. This is because
    columns with the same name, but different case, are still confusing
    and potentially ambiguous to other readers.

    In situations where it is *necessary* to have columns with the same
    name (whether they differ in case or not) we recommend disabling this
    rule for either just the line, or the whole file.
    """
```